File: RealLifeUi/Html/Componente/Missao/DivMissaoBase.cs
using NetZ.Web.Html.Componente;
using NetZ.Web.Server.Arquivo.Css;

namespace RealLifeUi.Html.Componente.Missao
{
    internal abstract class DivMissaoBase : ComponenteHtml
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = this.GetType().Name;
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            //this.addCss(css.setDisplay("none"));
            this.addCss(css.setMarginBottom(50));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: RealLiteUi.TypeScript/html/componente/missao/DivMissaoBase.ts
/// <reference path="../../../../Web.TypeScript/html/componente/ComponenteHtml.ts"/>

module RealLifeUi
{
    // #region Importações

    import ComponenteHtml = Web.ComponenteHtml;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export abstract class DivMissaoBase extends ComponenteHtml
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos
        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        protected inicializar(): void
        {
            super.inicializar();

            this.jq.css("margin-left", $(document).width());
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}